app/src/main/res/layout/item_playlist.xml

```diff
@@ -15,19 +16,76 @@
+        <RelativeLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content">
+
+            <LinearLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:orientation="vertical">
+
+                <TextView
+                    android:id="@+id/textView"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="10dp"
+                    android:text="@{song.artist}"
+                    tools:text="Artist" />
+
+                <TextView
+                    android:id="@+id/title"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:text="@{song.name}"
+                    tools:text="Name" />
+
+            </LinearLayout>
+
+            <ImageView
+                android:id="@+id/songplaying"
+                android:layout_width="32dp"
+                android:layout_height="32dp"
+                android:layout_alignParentRight="true"
+                android:layout_gravity="right"
+                android:layout_centerVertical="true"
+                android:src="@drawable/ic_play_arrow_accent_24dp"
+                android:visibility="@{song.songPlayingVisible}" />
+
+            <ImageView
+                android:id="@+id/songpaused"
+                android:layout_width="32dp"
+                android:layout_height="32dp"
+                android:layout_alignParentRight="true"
+                android:layout_gravity="right"
+                android:layout_centerVertical="true"
+                android:src="@drawable/ic_pause_accent_24dp"
+                android:visibility="@{song.songPausedVisible}" />
+
+            <ProgressBar
+                android:id="@+id/songloading"
+                style="@style/Widget.AppCompat.ProgressBar"
+                android:layout_width="32dp"
+                android:layout_height="32dp"
+                android:layout_alignParentRight="true"
+                android:layout_centerVertical="true"
+                android:indeterminate="true"
+                android:visibility="@{song.songLoadingVisible}" />
+
+            <com.github.lzyzsd.circleprogress.DonutProgress
+                android:id="@+id/songprogress"
+                android:layout_width="32dp"
+                android:layout_height="32dp"
+                android:layout_alignParentRight="true"
+                android:layout_centerVertical="true"
+                custom:donut_finished_stroke_width="3dp"
+                custom:donut_unfinished_stroke_width="3dp"
+                custom:donut_text_color="#00000000"
+                custom:donut_finished_color="#FE4083"
+                custom:donut_progress="@{song.stringProgress}"
+                android:visibility="@{song.songProgressVisible}" />
```

Comment:
i only see the inderminate progress bar the entire time. logcat is here, but i cant see any errors: https://gist.github.com/funkyg/e274aa116195df4d3aca716dc123a5f1